windows/hid.c

```diff
@@ -1150,51 +1228,38 @@ int HID_API_EXPORT HID_API_CALL hid_hotplug_register_callback(unsigned short ven
 		}
 	}
 
+	LeaveCriticalSection(&hid_hotplug_context.critical_section);
+
 	return 0;
 }
 
 int HID_API_EXPORT HID_API_CALL hid_hotplug_deregister_callback(hid_hotplug_callback_handle callback_handle)
 {
-	struct hid_hotplug_callback *hotplug_cb = NULL;
+	if (callback_handle <= 0 || !hid_hotplug_context.critical_section_ready) {
+		return -1;
+	}
 
-	if (callback_handle <= 0 || hid_hotplug_context.hotplug_cbs == NULL) {
+	/* Lock the mutex to avoid race conditions */
+	EnterCriticalSection(&hid_hotplug_context.critical_section);
+
+	if (!hid_hotplug_context.hotplug_cbs) {
+		LeaveCriticalSection(&hid_hotplug_context.critical_section);
 		return -1;
 	}
```

Comment:
This is actually done intentionally so that the hotplug_cbs pointer doesn't get the chance to be changed to NULL by any background thread (can happen even through an internal thread if the last callback returns a value that deregisters it)